From 335e9a5cb8fafd602d2fde906405234d94e0f753 Mon Sep 17 00:00:00 2001
From: Tatsuo ONISHI <onishi@latmos.ipsl.fr>
Date: Thu, 29 Sep 2022 19:38:38 +0200
Subject: [PATCH 24/70] (DEBUG) Aerosol Optical Properties

TYPE : Debug

modified files:
  chem/chemics_init.F
  dyn_em/start_em.F

Variables of aerosol properties without certain species like BC, SO4 etc
are included in arguments in subroutine chem_init

variables concerned:
  tauaer1_no_bc, tauaer2_.....
  gaer1_no_bc, gaer2_no_....
  waer1_no_bc, waer2_no_....
  tauaer1_no_so4, tauaer2_.....
  gaer1_no_so4, gaer2_no_....
  waer1_no_so4, waer2_no_....
  tauaer1_no_no3, tauaer2_.....
  gaer1_no_no3, gaer2_no_....
  waer1_no_no3, waer2_no_....
  tauaer1_no_nh4, tauaer2_.....
  gaer1_no_nh4, gaer2_no_....
  waer1_no_nh4, waer2_no_....
  tauaer1_no_oc, tauaer2_.....
  gaer1_no_oc, gaer2_no_....
  waer1_no_oc, waer2_no_....
---
 chem/chemics_init.F | 116 ++++++++++++++++++++++++++++++++++++++++++--
 dyn_em/start_em.F   |  25 +++++++++-
 2 files changed, 135 insertions(+), 6 deletions(-)

diff --git a/chem/chemics_init.F b/chem/chemics_init.F
index 4367dbc..b6bd7dd 100755
--- a/chem/chemics_init.F
+++ b/chem/chemics_init.F
@@ -7,12 +7,35 @@
 !lm 2015/07: diagnosis variables for accumulated BC deposition, dry_dep_bc and
 !wet_dep_bc, are initialized here to zero
 !
-   subroutine chem_init (id,chem,emis_ant,scalar,dt,bioemdt,photdt,chemdt,stepbioe, &
-               stepphot,stepchem,stepfirepl,plumerisefire_frq,z_at_w,xlat,xlong,    &
+   subroutine chem_init (id,chem,emis_ant,scalar,dt,bioemdt,photdt,  &
+!lm               chemdt,                                               &
+!lm               stepbioe,stepphot,stepchem,stepfirepl,                &
+               chemdt,radeffectdt,                                      &
+               stepbioe,stepphot,stepchem,stepradeffect,      &
+               stepfirepl,                                                   &
+!lm
+               plumerisefire_frq,z_at_w,xlat,xlong,    &
                g,aerwrf,config_flags,grid,alt,t,p,CONVFAC,ttday,tcosz,julday,gmt,   &
                tauaer1,tauaer2,tauaer3,tauaer4,                      &
                gaer1,gaer2,gaer3,gaer4,                              &
                waer1,waer2,waer3,waer4,                              &
+!lm
+               tauaer1_no_bc,tauaer2_no_bc,tauaer3_no_bc,tauaer4_no_bc, &
+               gaer1_no_bc,gaer2_no_bc,gaer3_no_bc,gaer4_no_bc, &
+               waer1_no_bc,waer2_no_bc,waer3_no_bc,waer4_no_bc, &
+               tauaer1_no_so4,tauaer2_no_so4,tauaer3_no_so4,tauaer4_no_so4, &
+               gaer1_no_so4,gaer2_no_so4,gaer3_no_so4,gaer4_no_so4, &
+               waer1_no_so4,waer2_no_so4,waer3_no_so4,waer4_no_so4, &
+               tauaer1_no_no3,tauaer2_no_no3,tauaer3_no_no3,tauaer4_no_no3, &
+               gaer1_no_no3,gaer2_no_no3,gaer3_no_no3,gaer4_no_no3, &
+               waer1_no_no3,waer2_no_no3,waer3_no_no3,waer4_no_no3, &
+               tauaer1_no_nh4,tauaer2_no_nh4,tauaer3_no_nh4,tauaer4_no_nh4, &
+               gaer1_no_nh4,gaer2_no_nh4,gaer3_no_nh4,gaer4_no_nh4, &
+               waer1_no_nh4,waer2_no_nh4,waer3_no_nh4,waer4_no_nh4, &
+               tauaer1_no_oc,tauaer2_no_oc,tauaer3_no_oc,tauaer4_no_oc, &
+               gaer1_no_oc,gaer2_no_oc,gaer3_no_oc,gaer4_no_oc, &
+               waer1_no_oc,waer2_no_oc,waer3_no_oc,waer4_no_oc, &
+!lm
                l2aer,l3aer,l4aer,l5aer,l6aer,l7aer,                  &
                extaerlw1,extaerlw2,extaerlw3,extaerlw4,              &
                extaerlw5,extaerlw6,extaerlw7,extaerlw8,              &
@@ -79,7 +102,9 @@
 
    IMPLICIT NONE
 
-   real  , intent(in) :: bioemdt,photdt,chemdt,dt,gmt
+!lm   real  , intent(in) :: bioemdt,photdt,chemdt,dt,gmt
+   real  , intent(in) :: bioemdt,photdt,chemdt,radeffectdt,dt,gmt
+!lm
    INTEGER,      INTENT(IN   ) :: plumerisefire_frq
    INTEGER,      INTENT(IN   ) :: chem_in_opt
    INTEGER,      INTENT(INOUT) :: num_vert_mix
@@ -119,6 +144,25 @@
                                dgnumwet_a1, dgnumwet_a2, dgnumwet_a3
 !-- end dgnum restart arrays
 
+!lm
+   REAL,  DIMENSION( ims:ime , kms:kme , jms:jme )         ,         &
+          INTENT(INOUT   ) ::                                        &
+                      tauaer1_no_bc,tauaer2_no_bc,tauaer3_no_bc,tauaer4_no_bc,      &
+                      gaer1_no_bc,gaer2_no_bc,gaer3_no_bc,gaer4_no_bc,              &
+                      waer1_no_bc,waer2_no_bc,waer3_no_bc,waer4_no_bc,              &
+                      tauaer1_no_so4,tauaer2_no_so4,tauaer3_no_so4,tauaer4_no_so4,      &
+                      gaer1_no_so4,gaer2_no_so4,gaer3_no_so4,gaer4_no_so4,              &
+                      waer1_no_so4,waer2_no_so4,waer3_no_so4,waer4_no_so4,              &
+                      tauaer1_no_no3,tauaer2_no_no3,tauaer3_no_no3,tauaer4_no_no3,      &
+                      gaer1_no_no3,gaer2_no_no3,gaer3_no_no3,gaer4_no_no3,              &
+                      waer1_no_no3,waer2_no_no3,waer3_no_no3,waer4_no_no3,              &
+                      tauaer1_no_nh4,tauaer2_no_nh4,tauaer3_no_nh4,tauaer4_no_nh4,      &
+                      gaer1_no_nh4,gaer2_no_nh4,gaer3_no_nh4,gaer4_no_nh4,              &
+                      waer1_no_nh4,waer2_no_nh4,waer3_no_nh4,waer4_no_nh4,              &
+                      tauaer1_no_oc,tauaer2_no_oc,tauaer3_no_oc,tauaer4_no_oc,      &
+                      gaer1_no_oc,gaer2_no_oc,gaer3_no_oc,gaer4_no_oc,              &
+                      waer1_no_oc,waer2_no_oc,waer3_no_oc,waer4_no_oc
+!lm
    REAL,  DIMENSION( ims:ime , kms:kme , jms:jme , 1:4 )   ,         &
           INTENT(INOUT   ) ::                                        &
                                l2aer,l3aer,l4aer,l5aer,l6aer,l7aer
@@ -141,7 +185,9 @@
           INTENT(INOUT   ) ::                                        &
                                ttday,tcosz,xlat,xlong
    real, INTENT (IN) :: g
-   integer, intent(out) :: stepbioe,stepphot,stepchem,stepfirepl
+!lm   integer, intent(out) :: stepbioe,stepphot,stepchem,stepfirepl
+   integer, intent(out) :: stepbioe,stepphot,stepchem,stepradeffect,stepfirepl
+!lm
    TYPE (grid_config_rec_type) , INTENT (in) ::     config_flags
    TYPE(domain) ,             INTENT (inout) ::     grid
    
@@ -558,6 +604,68 @@ ENDIF
                 waer2(i,k,j) = 0.
                 waer3(i,k,j) = 0.
                 waer4(i,k,j) = 0.
+!lm
+                tauaer1_no_bc(i,k,j) = 0.
+                tauaer2_no_bc(i,k,j) = 0.
+                tauaer3_no_bc(i,k,j) = 0.
+                tauaer4_no_bc(i,k,j) = 0.
+                gaer1_no_bc(i,k,j) = 0.
+                gaer2_no_bc(i,k,j) = 0.
+                gaer3_no_bc(i,k,j) = 0.
+                gaer4_no_bc(i,k,j) = 0.
+                waer1_no_bc(i,k,j) = 0.
+                waer2_no_bc(i,k,j) = 0.
+                waer3_no_bc(i,k,j) = 0.
+                waer4_no_bc(i,k,j) = 0.
+                tauaer1_no_so4(i,k,j) = 0.
+                tauaer2_no_so4(i,k,j) = 0.
+                tauaer3_no_so4(i,k,j) = 0.
+                tauaer4_no_so4(i,k,j) = 0.
+                gaer1_no_so4(i,k,j) = 0.
+                gaer2_no_so4(i,k,j) = 0.
+                gaer3_no_so4(i,k,j) = 0.
+                gaer4_no_so4(i,k,j) = 0.
+                waer1_no_so4(i,k,j) = 0.
+                waer2_no_so4(i,k,j) = 0.
+                waer3_no_so4(i,k,j) = 0.
+                waer4_no_so4(i,k,j) = 0.
+                tauaer1_no_no3(i,k,j) = 0.
+                tauaer2_no_no3(i,k,j) = 0.
+                tauaer3_no_no3(i,k,j) = 0.
+                tauaer4_no_no3(i,k,j) = 0.
+                gaer1_no_no3(i,k,j) = 0.
+                gaer2_no_no3(i,k,j) = 0.
+                gaer3_no_no3(i,k,j) = 0.
+                gaer4_no_no3(i,k,j) = 0.
+                waer1_no_no3(i,k,j) = 0.
+                waer2_no_no3(i,k,j) = 0.
+                waer3_no_no3(i,k,j) = 0.
+                waer4_no_no3(i,k,j) = 0.
+                tauaer1_no_nh4(i,k,j) = 0.
+                tauaer2_no_nh4(i,k,j) = 0.
+                tauaer3_no_nh4(i,k,j) = 0.
+                tauaer4_no_nh4(i,k,j) = 0.
+                gaer1_no_nh4(i,k,j) = 0.
+                gaer2_no_nh4(i,k,j) = 0.
+                gaer3_no_nh4(i,k,j) = 0.
+                gaer4_no_nh4(i,k,j) = 0.
+                waer1_no_nh4(i,k,j) = 0.
+                waer2_no_nh4(i,k,j) = 0.
+                waer3_no_nh4(i,k,j) = 0.
+                waer4_no_nh4(i,k,j) = 0.
+                tauaer1_no_oc(i,k,j) = 0.
+                tauaer2_no_oc(i,k,j) = 0.
+                tauaer3_no_oc(i,k,j) = 0.
+                tauaer4_no_oc(i,k,j) = 0.
+                gaer1_no_oc(i,k,j) = 0.
+                gaer2_no_oc(i,k,j) = 0.
+                gaer3_no_oc(i,k,j) = 0.
+                gaer4_no_oc(i,k,j) = 0.
+                waer1_no_oc(i,k,j) = 0.
+                waer2_no_oc(i,k,j) = 0.
+                waer3_no_oc(i,k,j) = 0.
+                waer4_no_oc(i,k,j) = 0.
+!lm
                 l2aer(i,k,j,1) = 0.
                 l2aer(i,k,j,2) = 0.
                 l2aer(i,k,j,3) = 0.
diff --git a/dyn_em/start_em.F b/dyn_em/start_em.F
index cf268c7..4f2f1bc 100644
--- a/dyn_em/start_em.F
+++ b/dyn_em/start_em.F
@@ -1825,8 +1825,12 @@ endif
          enddo
 
          CALL chem_init (grid%id,chem,emis_ant,scalar,grid%dt,grid%bioemdt,grid%photdt,     &
-                         grid%chemdt,                                       &
-                         grid%stepbioe,grid%stepphot,grid%stepchem,grid%stepfirepl,      &
+!lm                         grid%chemdt,                                       &
+!lm                         grid%stepbioe,grid%stepphot,grid%stepchem,grid%stepfirepl,      &
+                         grid%chemdt,grid%radeffectdt,                                      &
+                         grid%stepbioe,grid%stepphot,grid%stepchem,grid%stepradeffect,      &
+                         grid%stepfirepl,                                                   &
+!lm
                          grid%plumerisefire_frq,z_at_w,grid%xlat,grid%xlong,g,           &
                          grid%aerwrf,config_flags,grid,            &
                          grid%alt,grid%t_1,grid%p,convfac,grid%ttday,grid%tcosz,         &
@@ -1834,6 +1838,23 @@ endif
                          grid%tauaer1,grid%tauaer2,grid%tauaer3,grid%tauaer4,                   &
                          grid%gaer1,grid%gaer2,grid%gaer3,grid%gaer4,                           &
                          grid%waer1,grid%waer2,grid%waer3,grid%waer4,                           &
+!lm
+                         grid%tauaer1_no_bc,grid%tauaer2_no_bc,grid%tauaer3_no_bc,grid%tauaer4_no_bc, &
+                         grid%gaer1_no_bc,grid%gaer2_no_bc,grid%gaer3_no_bc,grid%gaer4_no_bc, &
+                         grid%waer1_no_bc,grid%waer2_no_bc,grid%waer3_no_bc,grid%waer4_no_bc, &
+                         grid%tauaer1_no_so4,grid%tauaer2_no_so4,grid%tauaer3_no_so4,grid%tauaer4_no_so4, &
+                         grid%gaer1_no_so4,grid%gaer2_no_so4,grid%gaer3_no_so4,grid%gaer4_no_so4, &
+                         grid%waer1_no_so4,grid%waer2_no_so4,grid%waer3_no_so4,grid%waer4_no_so4, &
+                         grid%tauaer1_no_no3,grid%tauaer2_no_no3,grid%tauaer3_no_no3,grid%tauaer4_no_no3, &
+                         grid%gaer1_no_no3,grid%gaer2_no_no3,grid%gaer3_no_no3,grid%gaer4_no_no3, &
+                         grid%waer1_no_no3,grid%waer2_no_no3,grid%waer3_no_no3,grid%waer4_no_no3, &
+                         grid%tauaer1_no_nh4,grid%tauaer2_no_nh4,grid%tauaer3_no_nh4,grid%tauaer4_no_nh4, &
+                         grid%gaer1_no_nh4,grid%gaer2_no_nh4,grid%gaer3_no_nh4,grid%gaer4_no_nh4, &
+                         grid%waer1_no_nh4,grid%waer2_no_nh4,grid%waer3_no_nh4,grid%waer4_no_nh4, &
+                         grid%tauaer1_no_oc,grid%tauaer2_no_oc,grid%tauaer3_no_oc,grid%tauaer4_no_oc, &
+                         grid%gaer1_no_oc,grid%gaer2_no_oc,grid%gaer3_no_oc,grid%gaer4_no_oc, &
+                         grid%waer1_no_oc,grid%waer2_no_oc,grid%waer3_no_oc,grid%waer4_no_oc, &
+!lm
                          grid%l2aer,grid%l3aer,grid%l4aer,grid%l5aer,grid%l6aer,grid%l7aer,     &
                          grid%extaerlw1,grid%extaerlw2,grid%extaerlw3,grid%extaerlw4,           &
                          grid%extaerlw5,grid%extaerlw6,grid%extaerlw7,grid%extaerlw8,           &
-- 
2.31.1

